AsyncDisplayKit/ASTextNode.mm

```diff
@@ -333,6 +338,8 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
   ASDisplayNodeAssert(constrainedSize.width >= 0, @"Constrained width for text (%f) is too  narrow", constrainedSize.width);
   ASDisplayNodeAssert(constrainedSize.height >= 0, @"Constrained height for text (%f) is too short", constrainedSize.height);
   
+  std::lock_guard<std::recursive_mutex> l(_textLock);
```

Comment:
Ahh, this is probably why it has to be recursive, we hold it while calling into methods that also lock.

If we can achieve this without a recursive mutex I'll be happier, since you're interacting with the global textkit locks inside the renderer, and as written I *think* there's a potential for lock inversion.